internal/venice-common/src/main/java/com/linkedin/venice/writer/VeniceWriter.java

```diff
@@ -1824,37 +1837,52 @@ private Segment startSegment(int partition) {
     }
   }
 
-  private void endAllSegments(boolean finalSegment) {
-    Segment segment;
+  private CompletableFuture<Void> endAllSegments() {
```

Comment:
The `finalSegment` always being `true`, is actually a bug. It should only be `true` when VW is closed. Can we keep the old interface? Fixing the bug may be out of scope, so, feel free to skip it (but it is low touch if you want to do it any way).

Reply:
Sure will add it back and fix the bug. Thanks

Reply:
Done